### PR TITLE
HardwareSet Component

### LINKED ISSUE #21: Hardware Capacity Component
React component to display the [hardware capacity](https://github.com/ece461l-fa24-final-project/hardwareby.us/issues/16) for a given hardware set for a given project.

### LINKED ISSUE #22: Hardware Availability Component
React component to display the [hardware availability](https://github.com/ece461l-fa24-final-project/hardwareby.us/issues/15) for a given hardware set for a given project.

### PR DESCRIPTION
**Pull request by DrewBritt**
React component to view the total capacity and current availability of an argument HardwareSet, as well as allowing checking in and out of said hardware set. Closes #21 and closes #22.

image

### CODE REVIEW

File: frontend/src/components/HardwareSetComponent.tsx
Comment (nuudlman) [inline frontend/src/components/HardwareSetComponent.tsx:1]: I would just call this file HardwareSet.tsx

File: frontend/src/utils/types.ts
Comment (nuudlman) [inline frontend/src/utils/types.ts:1]: You can just export this interface inside HardwareSet.tsx, but I doubt anyone else will use it.

File: frontend/src/components/HardwareSetComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                            marginRight: "10px",
+                        }}
+                    >
+                        Quantity:
+                        <input
+                            type="number"
+                            value={quantity}
+                            onChange={(e) =>
+                                setQuantity(
+                                    Math.max(1, parseInt(e.target.value) || 1),
```
Comment (nuudlman) [inline frontend/src/components/HardwareSetComponent.tsx:93]: What is this supposed to do?

Comment (nuudlman) [reply frontend/src/components/HardwareSetComponent.tsx:93]: Did you mean Math.min perchance?

**Review by nuudlman — CHANGES_REQUESTED** [review]
A few nits, but good work

File: frontend/src/components/HardwareSetComponent.tsx (hunk 1 above)
Comment (DrewBritt) [reply frontend/src/components/HardwareSetComponent.tsx:93]: It's acting as a min clamp at 1; either the max is 1 (so we have an input value such as 0 or a negative number) or the max is our input, meaning it's a (valid) number above 1.

**Review by nuudlman — APPROVED** [review]
LGTM

**Pull request merged**